bindings/python/samples/rotating-block-generator.py

```diff
@@ -34,25 +29,52 @@ def run(self):
 
         deg_to_rad = 2 * 3.14159265 / 360
         rotation = 0
+
+        # Pre calculate colors
+        col_table = []
+        for x in range(int(min_rotate), int(max_rotate)):
+            col_table.insert(x, scale_col(x, min_display, max_display))
+
+        # Pre calculate angles
+        sin_table = []
+        cos_table = []
+
+        for rotation in range(0, 360):
+            angle = rotation * deg_to_rad
+            sin_table.insert(rotation, math.sin(angle))
+            cos_table.insert(rotation, math.cos(angle))
+
         offset_canvas = self.matrix.CreateFrameCanvas()
 
         while True:
             rotation += 1
             rotation %= 360
 
+            sin = sin_table[rotation]
+            cos = cos_table[rotation]
+
             for x in range(int(min_rotate), int(max_rotate)):
                 for y in range(int(min_rotate), int(max_rotate)):
-                    ret = self.rotate(x - cent_x, y - cent_x, deg_to_rad * rotation)
-                    rot_x = ret["new_x"]
-                    rot_y = ret["new_y"]
+                    # translate so that the origin is now (cent_x, cent_y)
```

Comment:
This assignment to temp variables looks bit awkward here and hinders readability; I'd probably make a function to more emphasize what is happening here.
It would get `x`, `y` and pre-calculated `sin`, `cos` values returns a tuple with the new coordinates; roughly something like this:

```Python
new_x, new_y = rotate_coordinates(x - cent_x, y - cent_y, sin, cos)
new_x += cent_x
new_y += cent_y
```

The implementation of that function is of course trivial. But since this is an example, we should also make sure to suggest good coding practices for people starting out with programming.

Reply:
OK, I've moved to a method. It cost a few fps, about 17-18 now on my setup. I made the method `rotate_about_point` as that's more useful and gave me a place to explain the subtraction and addition operations that otherwise might appear magical.

I decided not to build the sin/cos lookup table. one sin and cos operation per frame makes no difference and simplifies the example. It still teaches the principle of pre-calculating through the colours, while not breaking if someone wants to do sub degree increments.